src/crimson/os/seastore/seastore_types.h

```diff
@@ -566,9 +567,22 @@ struct paddr_t {
   friend bool operator<(const paddr_t &, const paddr_t&);
   friend bool operator>=(const paddr_t &, const paddr_t&);
   friend bool operator>(const paddr_t &, const paddr_t&);
+  #define GET_DEV_ADDR(a)	\
+    (a & (std::numeric_limits<paddr_t::common_addr_t>::max() >> 1))
 };
 WRITE_EQ_OPERATORS_1(paddr_t, dev_addr);
-WRITE_CMP_OPERATORS_1(paddr_t, dev_addr);
+inline bool operator>(const paddr_t &l, const paddr_t &r) {
+  return GET_DEV_ADDR(l.dev_addr) > GET_DEV_ADDR(r.dev_addr);
```

Comment:
Why?  If the addr types are different, direct comparison doesn't make a ton of sense anyway.

Reply:
I added this part because I encountered ` assert(laddr >= ret.leaf.node->get_meta().begin && laddr < ret.leaf.node->get_meta().end);` in fixed_kv_btree.h. It seems that we need common MAX value to handle both seg and blk addr.

Reply:
> This is because current P_ADDR_MAX is make_seg_paddr(MAX_SEG_ID, MAX_SEG_OFF). 

Apart from the discussion about whether the 1-bit leading address type is necessary in paddr. Is it simpler and faster to instead define P_ADDR_MAX to be `paddr_t(std::numeric_limits<common_addr_t>::max())` ?

Reply:
Right. That is the simplest way. I thought that there was some reason P_ADDR_MAX is defined as make_seg_paddr(MAX_SEG_ID, MAX_SEG_OFF).